services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/protocol/serializer/AdminOperationSerializer.java

```diff
@@ -69,4 +104,46 @@ public static Map<Integer, Schema> initProtocolMap() {
       throw new VeniceMessageException("Could not initialize " + AdminOperationSerializer.class.getSimpleName(), e);
     }
   }
+
+  public static Schema getSchema(int schemaId) {
+    if (!PROTOCOL_MAP.containsKey(schemaId)) {
+      throw new VeniceMessageException("Admin operation schema version: " + schemaId + " doesn't exist");
+    }
+    return PROTOCOL_MAP.get(schemaId);
+  }
+
+  /**
+   * Serialize the object by writer schema
+   */
+  private <T> byte[] serialize(T object, Schema writerSchema, int writerSchemaId) {
+    try {
+      GenericDatumWriter<T> datumWriter = new GenericDatumWriter<>(writerSchema);
+      ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+      Encoder encoder = AvroCompatibilityHelper.newBinaryEncoder(byteArrayOutputStream, true, null);
+      datumWriter.write(object, encoder);
+      encoder.flush();
+      return byteArrayOutputStream.toByteArray();
+    } catch (IOException e) {
+      throw new VeniceMessageException(
+          "Could not serialize object: " + object.getClass().getTypeName() + " with writer schema id: "
+              + writerSchemaId,
+          e);
+    }
+  }
+
+  /**
+   * Deserialize the object from the writer schema to the reader schema, returning a GenericRecord.
+   */
+  private GenericRecord deserialize(byte[] serializedRecord, Schema readerSchema, int readerSchemaId) {
+    try {
+      GenericDatumReader<GenericRecord> datumReader = new GenericDatumReader<>(LATEST_SCHEMA, readerSchema);
+      InputStream in = new ByteArrayInputStream(serializedRecord);
+      BinaryDecoder decoder = AvroCompatibilityHelper.newBinaryDecoder(in, true, null);
+      return datumReader.read(null, decoder);
+    } catch (IOException e) {
+      throw new VeniceMessageException(
+          "Could not deserialize bytes back into GenericRecord object with reader version: " + readerSchemaId,
```

Comment:
Can we combine these two functions into one function called something like: `serializeWithDownConversion`?
These two functions are sharing the same names as the existing ones, which can cause confusions, and it seems no user will use the new functions separately, so I think consolidating them with a different name to make it less confusing.

Reply:
Discussed offline. We keep the `serialize` method to avoid duplication, while cut the `deserialize` to the main method. I have updated the code, please help check again. Thanks!